templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -687,6 +686,33 @@ CRM.$(function($) {
 {/literal}{/if}{literal}
 
 CRM.$(function($) {
+  onPaymentMethodChange();
+  $("#payment_instrument_id").on("change",function(){
+    onPaymentMethodChange();
+  });
+
+  function onPaymentMethodChange() {
+    var paymentInstrument = $('#payment_instrument_id').val();
```

Comment:
So, it seems the ability to have it on edit is kind of a side-effect of the fact the fields were added to the contribution form not the payment form - see this comment https://github.com/civicrm/civicrm-core/pull/10050#issuecomment-289431439